src/shared.js

```diff
@@ -795,8 +795,62 @@ function isNewline(node) {
   return node.type === "Newline";
 }
 
+function getSourceFromTSQualifiedName(node) {
+  let left;
+  let right;
+  switch (node.left.type) {
+    case "Identifier": {
+      left = node.left.name;
+      break;
+    }
+    case "TSQualifiedName": {
+      left = getSourceFromTSQualifiedName(node.left);
+      break;
+    }
+    default: {
+      left = ``;
+      break;
+    }
+  }
+  switch (node.right.type) {
+    case "Identifier": {
+      right = `.${node.right.name}`;
+      break;
+    }
+    default: {
+      right = ``;
+      break;
+    }
+  }
+  return left + right;
+}
+
+function getSourceFromModuleReference(node) {
+  switch (node.type) {
+    case "TSExternalModuleReference": {
+      return node.expression.value;
```

Comment:
As far as I can tell, both `typescript` and `@typescript-eslint/parser` allow _parsing_ any expression inside `require`: `import a = require(cool().thing)`. `typescript` then makes it an error later (during type checking?): “String literal expected.ts(1141)”.

So `.value` is not guaranteed to be a string here. I think it’s important that eslint-plugin-simple-import-sort doesn’t crash while you are writing your code and run it via your editor. For example, maybe someone doesn’t know that only string literals are allowed. Then eslint-plugin-simple-import-sort shouldn’t crash – that would be confusing. Instead it needs to do _something_ – not sure what.

Reply:
Fixed